[WIP]Instrumentation, test and fix repo for event lost issues

wenjiaswe opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/devel/pull-requests.md#the-pr-submit-process and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/devel/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/devel/pull-requests.md#write-release-notes-if-needed
4. If the PR is unfinished, see how to mark it: https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:
Create a public repo for instrumentation, test and fix contribution for event lost issues

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This issue is only for event lost issues investigation purpose.

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

Commit:
Add dynamic environment variable substitution to subpaths

bruceauyeung committed:
kubectl cp support colons-in-filename

Commit:
GC fallback to jsonmerge patch when SMP is not supported

vikaschoudhary16 committed:
Introduce priority class in the resource quota

Commit:
remove ipvs feature gateway

Commit:
Possible cipher suites values and tls versions in help for apiserver and kubelet

rramkumar1 committed:
Add ipvs module loading logic to gce scripts

sttts committed:
Update generated files

wenjiaswe commented:
cc @roycaihw @cheftako @yliaog @janetkuo @jpbetz , I opened this PR so we could all commit to the same branch.

roycaihw commented:
/ok-to-test

wenjiaswe commented:
Signed CLA as Google employee

wenjiaswe commented:
/retest

cheftako commented:
/cc @cheftako

wenjiaswe commented:
THIS IS ONLY USED TEMPORARILY FOR DEBUG PURPOSE.

@roycaihw @cheftako @jpbetz , I still have a list of TODO as shown in the last paragraph. Please help to see if there is anything missing or other suggestions.

 I have added event logging points in:
1. staging/src/k8s.io/apiserver/pkg/storage/wache_cache.go processEvent, this is where kube-apiserver updates cache with watched events (Record the events coming into apiserver)
2. staging/src/k8s.io/apiserver/pkg/storage/cacher.go sendWatchCacheEvent, this is where watcher processes the event and sends to the result channel  (Record the events going out of apiserver)
3. staging/src/k8s.io/client-go/tools/cache/reflector.go, watchHandler, it gets the events from result channel and watch using the resourceVersion (Record events trace in a couple of different place: 1. when first sync to the latest resource version, 2. apiserver startCaching, 3. shared by clients like scheduler, controller-manager).

At each point, SWAT is used as the label of logging, followed by file name and function name, then event type, meta.name, and event resourceVersion (this is monochronic for each event) are recored. Now each log entry looks like this:

For a pod creation: 
$ kubectl create -f https://github.com/kubernetes/website/blob/master/docs/user-guide/walkthrough/pod-nginx.yaml
Logs in kube-apiserver.log:

SWAT(reflector.go watchHandler) event type, name, resourceVersion: ADDED, nginx, 968  (Cacher calls reflector to get the resource version at the moment of call)
SWAT(watch_cache.go processEvent) event type, name, resourceVersion: ADDED, nginx, 968
SWAT(cacher.go sendWatchCacheEvent) event type, name, resourceVersion: ADDED, nginx, 968
SWAT(cacher.go sendWatchCacheEvent) event type, name, resourceVersion: ADDED, nginx, 968
SWAT(cacher.go sendWatchCacheEvent) event type, name, resourceVersion: ADDED, nginx, 968
SWAT(cacher.go sendWatchCacheEvent) event type, name, resourceVersion: ADDED, nginx, 968
SWAT(cacher.go sendWatchCacheEvent) event type, name, resourceVersion: ADDED, nginx, 968
SWAT(cacher.go sendWatchCacheEvent) event type, name, resourceVersion: ADDED, nginx, 968
SWAT(cacher.go sendWatchCacheEvent) event type, name, resourceVersion: ADDED, nginx, 968 (need to find a better place so no repeating messages)
SWAT(reflector.go watchHandler) event type, name, resourceVersion: ADDED, nginx, 968 (Cacher startCaching)

Logs in kube-scheduler.log:
SWAT(reflector.go watchHandler) event type, name, resourceVersion: ADDED, nginx, 968

Logs in kube-controller-manager.go:
SWAT(reflector.go watchHandler) event type, name, resourceVersion: ADDED, nginx, 968

**TODO**
1. As @cheftako suggested, record what exactly are the events that apiserver got and the events that apiserver sends out, so we know if there is discrepancy there. 
2. A simple comparer that compare the list of resourceVersions at different locations and see if there is any one missing. (No need to wait on this for running test, comparing tool is for easiness of analysis, as long as we have all the data points recorded, we can always look at it or use the comparer whenever available.)
3. Simplify log, now it takes too much memories. Maybe ok for now, depends on tests. If the test contains large clusters and lots of data points the memory issue would be big. 
4. Add data points in scheduler and controller-manager and see how they actually used the events info from reflector.
5. Working on etcd log parser to get the all the actual writing of event out. (Again, tests can be run with the data points as-is now, we don't need to wait for the etcd part, as all the info should be recorded in etcd data folder, as long as we save the file from the test runs, we can always look at it or use the parser whenever available.)

wenjiaswe commented:
/sig api-machinery

sjenning committed:
sysctls: create feature gate to track promotion

CaoShuFeng committed:
support AddAnnotation in admission attributes

rajansandeep committed:
create coredns and kube-dns folders

wenjiaswe commented:
cc @mml

roycaihw commented on staging/src/k8s.io/client-go/tools/cache/reflector.go:
```diff
@@ -392,6 +392,10 @@ loop:
 			newResourceVersion := meta.GetResourceVersion()
+			// SWAT Event lost: print out resourceVersion, EventType and event object name
+			// if event is for a pod or a node
+			fmt.Printf("SWAT,reflector/watchHandler,%s,%s,%s\n",
```
I see reflector being used in multiple components (both in server and clients). Can we add `reflector.name` in the log to identify which component we are looking at?

roycaihw reviewed (commented):
@wenjiaswe Thanks for the contribution. I left some comments. I understand that you are planning on switching to glog verbose level and some other name for this instrumentation, and also combining the similar logs for `cacher.watchers`.

roycaihw commented on staging/src/k8s.io/apiserver/pkg/storage/cacher.go:
```diff
@@ -888,6 +888,15 @@ func (c *cacheWatcher) sendWatchCacheEvent(event *watchCacheEvent) {
+		return
+	}
+	fmt.Printf("SWAT,cacher/sendWatchCacheEvent,%s,%s,%s\n",
+		watchEvent.Type, meta.GetName(), meta.GetResourceVersion())
```
It looks like most (if not all) of the `runtime.Object` carried by these events don't set the `TypeMeta`. As a result `watchEvent.Object.GetObjectKind().GroupVersionKind()` returns empty GVK. Don't know why that comes to happen (maybe apiserver doesn't need to know gvk when it's serving objects from etcd to clients). 

You could use `reflect.TypeOf()` to identify the Go type for now.

wenjiaswe replied:
OK, thanks. I will do that.

freehan committed:
add utils to patch pod status

freehan committed:
include patch permission for kubelets

freehan committed:
change kubelet status manager to use patch instead of put to update pod status

freehan committed:
add Patch support in fake kubeClient

freehan committed:
fix unit tests using Patch in fake client

freehan committed:
make update

dims committed:
Updated generated files

CaoShuFeng committed:
add WithAudit admission decorator

WithAudit admission decorator log annotations to audit events set by
the decorated admission controller

CaoShuFeng committed:
Log policy name from pod security policy

xmudrii committed:
apiextensions-apiserver: add establishing controller to avoid race between established and CRs actually served

mlmhl committed:
implement kubelet side online file system resize for volume

vikaschoudhary16 committed:
Auto-generated files

vikaschoudhary16 committed:
Add probe based mechanism for kubelet plugin discovery

smarterclayton committed:
Update pod phase documentation

jiayingz committed:
Add gpu cluster upgrade test.

ericchiang committed:
core v1: deprecate the gitRepo volume type

ericchiang committed:
generated: update generated API files

Commands run:

	./hack/update-api-reference-docs.sh
	./hack/update-generated-protobuf.sh
	./hack/update-generated-swagger-docs.sh
	./hack/update-openapi-spec.sh
	./hack/update-swagger-spec.sh

enj committed:
Correctly apply request transforms with flattened resource builder

This change moves the NewClientWithOptions call into
Builder.getClient.  Since getClient is the only way for Builder and
its visitors to create a RESTClient, we can reasonably guarantee
that the request transforms will be honored.  Previously, it was
possible for a call to NewFlattenListVisitor to return resource Info
objects whose Client field did not honor the request transforms.

Signed-off-by: Monis Khan <mkhan@redhat.com>

vladimirvivien committed:
CSI implementation of raw block volume support

soltysh committed:
Never clean backoff in job controller

jhorwit2 committed:
Add support for enforcing read only host paths in PSPs.

chrisohaver committed:
dns record scale test

ingvagabund committed:
Promote sysctl annotations to API fields

wgliang committed:
use subtest for table units (pkg/master)

hzxuzhonghu committed:
Support dynamicly set logging verbosity

mikedanese committed:
implement service account token projection

dims committed:
API updates for Cinder Volumes to support for user specified Secrets in the future

mikedanese committed:
add api for service account token volume projection

mikedanese committed:
autogenerated

wgliang committed:
Move pkg/scheduler/schedulercache -> pkg/scheduler/cache

losipiuk committed:
Register RequestedToCapacityRatioPriority priority function

Random-Liu committed:
Remove direct and indirect streaming runtime interface.

janetkuo committed:
Test job backoffLimit correctly

mikedanese committed:
svcacct: validate min and max expiration seconds on TokenRequest

Random-Liu committed:
Proxy container streaming in kubelet.

Random-Liu committed:
Update bazel.

feiskyer committed:
Add security context for Windows containers

feiskyer committed:
Setup windows security context in CRI

feiskyer committed:
Setup docker options according to windows security context

freehan committed:
add ReadinessGates in pod spec

freehan committed:
make update

freehan committed:
validation and feature gate

Random-Liu committed:
Update unit test.

ingvagabund committed:
Switch to Beta

Commit:
Refactor of GenerateMapDeviceFunc to delegate Map call to volume plugin.

losipiuk committed:
Allow parametrization of RequestedToCapacityRatio priority function via policy config

freehan committed:
add utils for pod condition

mtaufen committed:
add a flag to control the cap on images reported in node status

While I normally try to avoid adding flags, this is a short term
scalability fix for v1.11, and there are other long-term solutions in
the works, so we shouldn't commit to this in the v1beta1 Kubelet config.
Flags are our escape hatch.

lichuqiang committed:
API changes for Topology aware dynamic provisioning

lichuqiang committed:
Provision interface change

Commit:
fix field removal in mutating admission webhooks

A mutating admission controller webhook doesn't remove object fields
when instructed to.

E.g. when the JSON patch

  [
    {"op": "remove", "path": "/spec/containers/0/resources/limits/fpga-arria10"},
    {"op": "add", "path": "/spec/containers/0/resources/limits/fpga-interface-id-524abcf", "value": 1}
  ]

is applied to this pod

apiVersion: v1
kind: Pod
metadata:
  name: test-pod
spec:
  restartPolicy: Never
  containers:
    -
      name: test-pod-container
      image: ubuntu:bionic
      imagePullPolicy: IfNotPresent
      command: [ "ls", "-l", "/" ]
      resources:
        limits:
          fpga-arria10: 1

in order to replace the resource name "fpga-arria10" with something understandable
by the device plugin the resulting pod spec still contains the old field plus
a new one. The resulting pod looks like

apiVersion: v1
kind: Pod
metadata:
  name: test-pod
spec:
  restartPolicy: Never
  containers:
    -
      name: test-pod-container
      image: ubuntu:bionic
      imagePullPolicy: IfNotPresent
      command: [ "ls", "-l", "/" ]
      resources:
        limits:
          fpga-arria10: 1
          fpga-interface-id-524abcf: 1

The patch unmarshals patched JSON into a new empty object instead of
existing one. Otherwise JSON unmarshaling reuses existing maps, keeping
existing entries as specified in the "encoding/json" standard package.

gnufied committed:
Implement API changes needed for dynamic volume limits

define alpha feature and make api changes

gnufied committed:
Implement scheduler changes for volume limits

gnufied committed:
Implement volume plugin changes for volume limits

gnufied committed:
Implement kubelet side changes for writing volume limit to node

Add tests for checking node limits

verult committed:
Refactored disk cloudprovider methods to use generated client; Refactored gce_disks unit tests; Removed unused gce_op.go and associated unit tests.

verult committed:
Generated code for gce_disks refactor

mtaufen committed:
document per-field advice for dynamic Kubelet config

deads2k committed:
remove unnecessary factory delegation for RESTClientGetter method

kad committed:
UX improvement for preflight check for external etcd client certificates

By using same preflight check label for all etcd client certificates, it will
allow user to use single identifier, and as result shorter command line
arguments to kubeadm, in case multiple issues found with those files.

Fixes: kubernetes/kubeadm#834

wangzhen127 committed:
Use default seccomp profile for GCE manifests

juanvallejo committed:
move resource builder flags to genericclioptions

juanvallejo committed:
move filename flags to genericclioptions

sttts committed:
client-go: start fresh with owner file

krzysied committed:
Fixing scheduling latency metrics

freehan committed:
add ContainersReady condition

dims committed:
Deprecate the in-tree keystone plugin

We now have the `client-keystone-auth` in cloud-provider-openstack
repository:
https://github.com/kubernetes/cloud-provider-openstack/blob/master/docs/using-client-keystone-auth.md

So let's drop the in-tree one, so we can remove it in 1.12

neolit123 committed:
kubeadm: conditionally set the kubelet cgroup driver for Docker

Add a new utility file - util/cgroupdriver.go.
Currently it only contains the public function GetCgroupDriverDocker().
The function uses 'docker info' to obtain the cgroup driver
for Docker.

On a later stage this file can contain more methods for different
CRI.

Use GetCgroupDriverDocker() in phases/kubelet/flags.go
to conditionally set the 'cgroup-driver' argument. On error
print a warning and don't set the argument value.

Add unit tests in cgroupdriver_test.go.

msau42 committed:
Only mount subpath as readonly if specified in volumeMount

ravisantoshgudimetla committed:
Phase out rescheduler in favor of priority and preemption

idealhack committed:
apiserver: update tests to use sub-benchmarks (aes_test.go)

idealhack committed:
apiserver: update tests to use sub-benchmarks (secretbox_test.go)

CaoShuFeng committed:
fix kubectl -o

Fix kubectl -o error message:
Before this change:
```shell
kubectl get pods -o foo
error: unable to match a printer suitable for the output format "" and the options specified: &get.PrintFlags{JSONYamlPrintFlags:(*genericclioptions.JSONYamlPrintFlags)(0x23aa610), NamePrintFlags:(*genericclioptions.NamePrintFlags)(0xc42058b4e0), TemplateFlags:(*printers.KubeTemplatePrintFlags)(0xc4206765e0), CustomColumnsFlags:(*printers.CustomColumnsPrintFlags)(0xc420676620), HumanReadableFlags:(*get.HumanPrintFlags)(0xc4204eb180), NoHeaders:(*bool)(0xc4206fefbc), OutputFormat:(*string)(0xc42058b4d0)}
```

After this change:
```shell
Kubectl get pods -o foo
error: unable to match a printer suitable for the output format "aaa", allowed formats are: custom-columns,custom-columns-file,go-template,go-template-file,json,jsonpath,jsonpath-file,name,template,templatefile,wide,yaml
```

Commit:
Declare wait flag in way consistent with other deletion flags

lichuqiang committed:
Dynamic provisioning allowed topologies scheduler work

andyzhangx committed:
azuredisk size grow feature

fix comments

fix comments

pohly committed:
e2e/storage: central argument handling

Putting the command line argument handling into the central test
context seems like the better solution, in particular considering that
argument handling might get changed in the future to use Viper.

shyamjvs committed:
Fix bug with scheduler throughput variable pass-by-value

rosti committed:
kubeadm: Use loadPodSpecFromFile instead of LoadPodFromFile

Implement and use loadPodSpecFromFile which loads and returns PodSpec object
from YAML or JSON file. Use this function in the places where LoadPodFromFile
is used to load Pod object and then return the PodSpec portion of it. This
also removes the dependency on //pkg/volume/util and its dependencies (thus,
making kubeadm more lean).

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

wojtek-t committed:
Fix GKE Regional Clusters upgrade tests

soltysh committed:
Increase the timeout when waiting for the job to be gone

jsafrane committed:
Allow AWS EBS volumes to be attached as ReadOnly.

soltysh committed:
DaemonSet internals are still in extensions

soltysh committed:
Add daemonset when to getReplicasFromRuntimeObject when cleaning objects in e2e

sttts committed:
client-go: document README exception in .github/PULL_REQUEST_TEMPLATE.md

wojtek-t committed:
Parallelize taint manager

jsafrane committed:
Add block volume support to internal provisioners.

dixudx committed:
include rollout object name in cli message

deads2k committed:
services must listen on port 443

deads2k committed:
fix the delete result being used

deads2k committed:
cleanup some dead kubectl code and narrow scope of helpers

jsafrane committed:
e2e test for block volume provisioning

deads2k committed:
add resource builder flags

luxas committed:
kubeadm: Move .NodeName and .CRISocket to a common sub-struct

luxas committed:
Update unit tests to use the new NodeRegistration object

luxas committed:
autogenerated

deads2k committed:
update set selector to use resource builder flags

yujuhong committed:
node e2e: fix the missing square brackets

Also tag the inode eviction tests with [NodeFeature:Eviction].

dashpole committed:
fix dynamic kubelet config tests

Commit:
Add dry-run to auth reconcile

Signed-off-by: Matt Rogers <mrogers@redhat.com>

nberlee committed:
Run CoreDNS container only with CAP_NET_BIND_SERVICE, drop all other (root) privileges.
 Run filesystem of container and config in read-only mode.

yujuhong committed:
e2e node: mark pod cgroup test as [NodeConformance]

liztio committed:
Save kubeadm manifest backup directories

When kubeadm upgrades a static pod cluster, the old manifests were previously
deleted. This patch alters this behaviour so they are now stored in a
timestamped temporary directory.

verult committed:
Modified regional PD test to fetch template name from GCE

Commit:
Add metrics for envelop transformer:
    transformation_operation_count
    transformation_failures_count
    envelope_transformation_cache_misses_count
    data_key_generation_latencies_microseconds
    data_key_generation_failures_count

ravisantoshgudimetla committed:
Build files generated

mtaufen committed:
remove unused status per TODO

This should have been deleted in #63221, as it is now unused.

MrHohn committed:
[gce provider] More wrappers for alpha/beta backend service

luxas committed:
kubeadm: Add a 'kubeadm config migrate' command

luxas committed:
autogenerated

Commit:
Promote watch e2e test to conformance

hanxiaoshuai committed:
remove unused parameter in func buildFakeProxier

Commit:
Add Vertical Pod Autoscaling API to the autoscaling group.

soltysh committed:
Fix error message to be consistent with others

Commit:
CSI fix for gRPC conn leak, test updates

Commit:
Auto-generated code for the Vertical Pod Autoscaler API.

Commit:
Add validation code for the Vertical Pod Autoscaler API.

Commit:
Add TLS support to exec authenticator plugin

https://github.com/kubernetes/community/blob/master/contributors/design-proposals/auth/kubectl-exec-plugins.md#tls-client-certificate-support

Allows exec plugin to return raw TLS key/cert data. This data populates
transport.Config.TLS fields.
transport.Config.TLS propagates custom credentials using
tls.Config.GetClientCertificate callback.
On key/cert rotation, all connections using old credentials are
closed

ostromart committed:
Update istio addon manifest to 0.8

tallclair committed:
Validate git args are not flags prior to mounting

hanxiaoshuai committed:
remove unused code in kubeadm error.go

hanxiaoshuai committed:
kubeadm uses its own scheme instead of kubectl scheme

sttts committed:
apiextensions: add ObjectMeta schema validation and pruning

ingvagabund committed:
Mount the kubeletConfigPath rw when running containerized node e2e tests

The kubelet needs to create dynamic-kubelet-config directory under the kubeletConfigPath
when initialing dynamic config directory.

andrewsykim committed:
cloud node controller: improve error handling for node registration

ncdc committed:
code-gen: support running from anywhere

Support running generate-groups.sh from any directory, so users can run
it from their own projects, instead of forcing them to run it from here.
This allows the generators to find and use go source code from a
project's vendor directory, for example.

Signed-off-by: Andy Goldstein <andy.goldstein@gmail.com>

luxas committed:
Move helper funcs and constants to the client-go Bootstrap Token package from kubeadm

luxas committed:
kubeadm: Initial refactor of the Bootstrap Tokens. Add the new API objects, add/move helpers and start using the new flow in the code

luxas committed:
Add unit tests for the new Bootstrap Token objects and functions

luxas committed:
autogenerated

tallclair committed:
Add tallclair to milestone maintainers

liztio committed:
Kubeadm/k8s version mismatch is now a skippable error

juanvallejo committed:
remove extra "../" when copying from pod to local

tallclair committed:
GitRepo command hardening

Commit:
openapi: Remove FakeClient from testing library

And make a simplified version of it where needed.

Commit:
Replace openapi Fake with kube-openapi version

There are two version of this fake class, let's just use that one
version.

ericchiang committed:
clientauthentication: add v1beta1 API version

gnufied committed:
Rename online resizine feature gate

ericchiang committed:
client-go: promote exec plugin support to beta

ericchiang committed:
generated:

	./hack/update-bazel.sh
	./hack/update-codegen.sh
	./hack/update-staging-godeps.sh

dashpole committed:
disable memcg for testing prior to 1.11 release

Commit:
Quote shell variable expansion

Commit:
Update CHANGELOG-1.11.md for v1.11.0-beta.1.

liggitt committed:
add debugging for aggregator flake

hanxiaoshuai committed:
fix bug excludeCIDRs was not assign in func NewProxier

ravisantoshgudimetla committed:
Revert "Remove rescheduler and corresponding tests from master"

cadmuxe committed:
Add netd as an addon for GKE.

vikaschoudhary16 committed:
Add feature gate for kubelet plugin watcher

dixudx committed:
apply global flag "context" for kubectl config view

Commit:
Increase timeout

chuckha committed:
Replace glog.Info{f,ln} with fmt.Print{f,ln}

This follows the pattern `kubectl` uses for logging.

There are two remaining glog.Infof call that cannot be removed easily.

One glog call comes from kubelet validation which calls features.SetFromMap.
The other comes from test/e2e during kernel validation.

Mostly fixes kubernetes/kubeadm#852

Signed-off-by: Chuck Ha <ha.chuck@gmail.com>

luxas committed:
fix the verify job

ericchiang committed:
client-go: make exec auth and auth provider mutually exclusive

sttts committed:
printers: fix json types  – int64 is only allowed integer

We have the invariant in apimachinery that all integers in JSON are int64.
We panic on other types on deepcopy and possibly at other occasions.

sttts committed:
conversions: don't mutate in.ObjectMeta.Annotations

sttts committed:
apimachinery: adapt ObjectConvertor invariant

We avoid deepcopies in the codec/conversion stack by re-using data structures.
This means that the out object of a conversion must be deepcopied before mutation
in order to not mutate the in object as well.

This guarantees that e.g. runtime.Encode (which uses conversion from internal -> versioned)
does not mutate the input. This would be highly unexpected (and we do not mention possible
mutation of the input for runtime.Encode).

liggitt committed:
Add wait.PollImmediateUntil

liggitt committed:
Avoid deadlock in gc resync if available resources change during sync

ostromart committed:
Set deployment security profile to docker/default

Random-Liu committed:
Address comments in #64006.

Signed-off-by: Lantao Liu <lantaol@google.com>

k82cn committed:
Updated helper funcs to use nodename.

k82cn committed:
Eanbled schedule DaemonSet Pods by default scheduler.

k82cn committed:
Updated integration test.

freehan committed:
Generate pod ready status with readiness gates

freehan committed:
trigger kubelet sync pod on reconciliation

freehan committed:
Generate ContainersReady condition

freehan committed:
Inject ContainersReady

sttts committed:
printers: add deepcopy tests to generated tables

liggitt committed:
Clarify --hostname-override and --cloud-provider interaction

stealthybox committed:
kubeadm: Set the kubelet `--resolv-conf` flag conditionally on init

Fixes https://github.com/kubernetes/kubeadm/issues/845

cadmuxe committed:
support netd on k8s

stealthybox committed:
kubeadm: When etcd is listening on all interfaces, set the etcd probe to use loopback

Fixes https://github.com/kubernetes/kubeadm/issues/882

cadmuxe committed:
Code clean up

cadmuxe committed:
add NON_MASTER_NODE_LABELS to config-test.sh

dixudx committed:
kubeadm: use nodeSelector to only run kube-proxy on architecture consistent nodes

dixudx committed:
kubeadm: use nodeselector instead of affinity in kube-dns

sttts committed:
apimachinery: do not fuzz ObjectMeta.{Labels/Annotation} with empty keys

Commit:
Remove event handler to satisfy alpha tests

liztio committed:
Bazel artefacts

mtaufen committed:
Fix test tag on dynamic config tests

The test accidentally got turned off when the NodeAlphaFeature tag was
added in #64125. This PR updates the tag to turn it back on.

liztio committed:
Kubeadm-initialised kubelet uses provided hostname if present

If --node-name is passed in to `kubeadm init`, `--hostname-override` will be
passed to kubelet.

gnufied committed:
Fix panic caused by no cloudprovider in test

We should not panic when no cloudprovider is present

chuckha committed:
Fix kubeadm for v1alpha1 configs

Signed-off-by: Chuck Ha <ha.chuck@gmail.com>

davidz627 committed:
Update container-storage-interface/spec vendor to v0.3.0 (and related dependencies)

rajansandeep committed:
coredns to gcr.io repo

sbezverk committed:
Adding CSI driver registration

davidz627 committed:
update golang/protobuf to v1.1.0 to satisfy CSI v0.3.0 hard requirement

sbezverk committed:
New labelmanager package

liztio committed:
kubeadm lowercases all domain names passed as additional SANs

Some domains, like ELBs, output a domain name with uppercase letters. To
accept these, we lowercase all arguments passed to ----apiserver-cert-extra-sans

silveryfu committed:
Update generated bazel

silveryfu committed:
Fix the handling of untagged images

ostromart committed:
Modify security profile for proxy

davidz627 committed:
Bazel BUILD fixes

yguo0905 committed:
Update GCE cloud provider to use Cloud TPU v1 API

yguo0905 committed:
Update dependency

mbohlool committed:
Add mbohlool to apiextensions-apiserver reviewer list

davidz627 committed:
staging godep update

soltysh committed:
Rate limit only when an actual error happens, not on update conflicts

jiayingz committed:
Reconcile extended resource capacity after kubelet restart.

jingxu97 committed:
Set GCE PD attachable volume limit based on machineType

This PR implements the function to return attachable volume limit based
on machineType for GCE PD. This is part of the design in kubernetes/community#2051/

lichuqiang committed:
generated files

CaoShuFeng committed:
remove deprecated option '--enable-custom-metrics'

feiskyer committed:
Fix panic while provisioning Azure security group rules

losipiuk committed:
Create new variable for each iteration step

Random-Liu committed:
Fix standalone dockershim.

vikaschoudhary16 committed:
Fix quota sync

rosti committed:
kubeadm: Don't match DNS versions to K8s versions

Some code in kubeadm was designed with the intent, that in the future CoreDNS
and kube-dns versions will match to specific K8s versions. This code is not
functional, since it does not perform any version matching. As of this moment,
no version matching is planned and a lot of boilerplate code is left useless.
The solution is simple - remove the unneeded parts to simplify the flow.

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

ghuntley committed:
housekeeping: improved language used in ISSUE_TEMPLATE.md

nikhita committed:
apiextensions: allow Description in the root schema for subresources

rajansandeep committed:
coredns to use gcr.io repo

ingvagabund committed:
Run make update

luxas committed:
kubeadm: Update the dropin for the kubelet in v1.11

luxas committed:
kubeadm: Upload CRISocket information and hence make kubeadm join blocking

luxas committed:
autogenerated

mikedanese committed:
auth: standalone kubelets shouldn't start a token manager

janetkuo committed:
Skip updating status for DaemonSet being deleted

janetkuo committed:
Remove hack in kubectl delete that handles DaemonSet deletion

dashpole committed:
update cadvisor godeps to v0.30.0

dashpole committed:
disable process scheduler metrics

Commit:
Update CHANGELOG-1.10.md for v1.10.4.

chuckha committed:
Adds a crictl package for kubeadm installs

Closes kubernetes/kubeadm#811

Signed-off-by: Chuck Ha <ha.chuck@gmail.com>

sttts committed:
Remove myself from sig-cli OWNER alias

As I am not active in that area right now. Avoids some noise.

janetkuo committed:
Revert "Remove hack in kubectl delete that handles DaemonSet deletion"

This reverts commit 10a12ddb347a98f97e9f4c9eb35bc9b8d59384fa.

janetkuo committed:
Add TODO for removing kubectl DaemonSet deletion hack

gnufied committed:
Fix setup of configmap/secret/projected/downwardapi

Only call setup after they are found; otherwise
we are left with orphan directories that are never
cleaned up.

luxas committed:
Add a 'kubeadm upgrade node config' command and finish up the kubelet integration work

luxas committed:
autogenerated

chuckha committed:
Use repo prefix when generating image names

CI defines its own custom repository. The function responsible
for listing all images now takes this into account.

Closes kubernetes/kubeadm#901

Signed-off-by: Chuck Ha <ha.chuck@gmail.com>

Commit:
Update CHANGELOG-1.11.md for v1.11.0-beta.2.

liggitt committed:
Revert "Add Vertical Pod Autoscaling API to the autoscaling group."

This reverts commit 3f92d3fcda583dd6ac872400609d21dcfa7fb643.

liggitt committed:
Revert "Auto-generated code for the Vertical Pod Autoscaler API."

This reverts commit da65f30e2aca4ca8177fcf27e95909e2883b6e20.

liggitt committed:
Revert "Add validation code for the Vertical Pod Autoscaler API."

This reverts commit 390cfec6172fb3efdb5e13b96dcf942bc7bf2124.

chuckha committed:
Fix output of `kubeadm migrate config`

The output should always be valid kubeadmapi.MasterConfiguration YAML.

Signed-off-by: Chuck Ha <ha.chuck@gmail.com>

dims committed:
Ensure directory is created for kubelet configuration

caseydavenport committed:
Don't specify a description for Calico CRDs

mfojtik committed:
improve memory footprint of daemonset simulate

aleksandra-malinowska committed:
Cluster Autoscaler 1.3.0-beta.1

fabriziopandini committed:
fix-kubeadm-pull-log

wenjiaswe committed:
instrumentation update

wenjiaswe commented:
@roycaihw new update pushed.

wenjiaswe commented:
/remove-sig scheduling
